test/common/http/date_provider_impl_test.cc

```diff
@@ -11,11 +11,63 @@
 #include "gtest/gtest.h"
 
 using testing::NiceMock;
+using testing::Return;
```

Comment:
Can you add a simple integration test that shows this works end-to-end, i.e from proto config to data plane request? I think we only need a single test here.

Reply:
I added a test to the router [here](https://github.com/envoyproxy/envoy/pull/11915/files#diff-6eb802c7b8ea53fcfa4c3f2842b28c7bR6567-R6645), is there something else you'd like covered that isn't covered by that test?

Reply:
Maybe something similar to this test, on the e2e level: 
https://github.com/envoyproxy/envoy/blob/706d9761dd1336ce0b650b34dae649312ec85c0b/test/integration/http_integration.cc#L717

Reply:
Done! Note that because the HTTP integration test provider doesn't currently provide functionality for overriding the current time, we're limited to only validating the presence of the `date` header (rather than its value)